### ISSUE TITLE
Provide guidance for Mathematics for Computer Science.

### ISSUE THREAD

**Issue by xxylem** [ossu/computer-science#591]
The 2015 MIT course Mathematics for Computer Science provides no solutions. Per #405, this is an old issue going back to 2017 and something that wasn't dealt with. Students need meaningful feedback for learning to happen.

My solution doing the course has been to do the following:

[2015 Online Lecture videos](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-spring-2015/proofs/)

[2015 Readings](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-spring-2015/readings/) 

[2010 Recitation Problems with Solutions](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-fall-2010/recitations/)

[2005 Problem Sets/Assignments](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-fall-2005/assignments/)

[2010 Video Lectures](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-fall-2010/video-lectures/)

The problem sets and recitation problems from the older courses have solutions and have mostly followed the same path as the newer course so far with the occasional reference to something unknown (still, this is better than no solutions). Do you have experience with this course and dealing with this issue? What did you do to solve this?

I think we should provide a bit more guidance to help future students. Something like the above could work.

**Comment by waciumawanjohi** [comment]
@xxylem 
I like your approach to the recitation and problem sets. I notice that you have 2 sets of video lectures. I hesitate to recommend that students take twice the time watching lectures, when one set should suffice.

If you were to recommend one lecture set, which would it be?

**Comment by xxylem** [comment]
Yeah I would put the 2015 lectures as recommended and the older ones as purely optional. I'm creating an advice page at the moment. I will add this advice to it.

**Comment by xxylem** [comment]
I have added this to the pull request #596. If you have any questions or comments, please add them there.

Thanks,

Josh

**Comment by waciumawanjohi** [comment]
Josh,
The 2015 CS for Math course [lists Single Variable Calc as a pre-requisite](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-spring-2015/syllabus/) for the course. Did you find that calculus was necessary to follow the lectures or to complete the recitation/problem sets that you recommend?

**Comment by annateng** [comment]
I'm working through the curriculum and encountering this same problem now. I see there's a Fall 2010 version of the course available online, including exam solutions and problem set solutions.

Mathematics for Computer Science Fall 2010: https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-042j-mathematics-for-computer-science-fall-2010/index.htm

Fall 2010 Problem Set Solutions: https://www.quora.com/Where-can-I-find-solutions-to-6-042-fall-2010-problem-sets

Is there any reason not to replace the course in its entirety with the Fall 2010 version?

**Comment by xxylem** [comment]
@waciumawanjohi Sorry, I didn't see your comment! I didn't find calculus to be particularly necessary for doing this course, only really the general approach to problem solving I learned from doing calculus was useful.

@annateng Personally, I found the 2015 video lectures to be better, it's just the professor talking directly to you and explaining things, whereas the 2010 videos are in a lecture hall to a full room. I prefer the former style. But yes, I did the 2010 problem sets/exams/recitations since for the newer ones no solutions are available.
